src/lti_toolbox/lti.py

```diff
@@ -162,3 +168,44 @@ def is_edx_format(self):
 
         """
         return re.search(r"^course-v[0-9]:.*$", self.get_param("context_id"))
+
+    def _has_any_of_roles(self, roles: Set[str]):
+        """Check if the LTI user has any of the provided roles."""
+        return not roles.isdisjoint(self.roles)
+
+    @property
+    def is_student(self):
+        """Check if the LTI user is a student.
+
+        Returns:
+            boolean: True if the LTI user is a student.
+
+        """
+        return self._has_any_of_roles({LTIRole.STUDENT, LTIRole.LEARNER})
+
+    @property
+    def is_instructor(self):
+        """Check if the LTI user is an instructor.
+
+        Returns:
+            boolean: True if the LTI user is an instructor.
+
+        """
+        return self._has_any_of_roles(
+            {LTIRole.INSTRUCTOR, LTIRole.TEACHER, LTIRole.STAFF}
+        )
+
+    @property
+    def is_administrator(self):
+        """Check if the LTI user is an administrator.
+
+        Returns:
+            boolean: True if the LTI user is an administrator.
+
+        """
+        return self._has_any_of_roles({LTIRole.ADMINISTRATOR})
+
+    @property
+    def can_edit_content(self):
+        """Check if the LTI user can edit LMS content."""
+        return self.is_administrator or self.is_instructor
```

Comment:
I don't know if this should be part of the library. In marsha we choose to allow both instructor and administrator and probably this will be true in most of LTI integration but this is most a choice made by the developer using this library.

Reply:
Another thing came to my mind: roles are not really part of the specifications 
https://www.imsglobal.org/specs/ltiv1p0/implementation-guide#toc-9
> Some of the commonly used roles from LIS include Learner, Instructor, Administrator, TeachingAssistant, ContentDeveloper, and Mentor. 

So, maybe we should have a setting for each of our roles:
`ROLES_FOR_STUDENT`, `ROLE_FOR_INTRUCTOR`, etc.
This way the library users could adapt to each LMS configuration.

Reply:
Also, it is worth mentioning that the spec expect simple TP to:

>  TC systems should be aware that simple TPs will key off the presence or absence of the Instructor role and group users into those with the Instructor role (read-write-configure) and those without the Instructor role (read).

Reply:
After discussing it with @jmaupetit, currently if a developer wants another set of roles allowed to edit content, they can overload this property, and make their own choice. I feel it great to have an opinionated way to define who can edit content for now, to assure a coherent choice across different fun's projects.

Reply:
After a discussion with @jmaupetit, the current approach allows developers to overload this property and make their own choice if they want another set of roles to be able to edit content.However, I believe it would be beneficial to introduce an opinionated method for defining who can edit content, ensuring a consistent choice across various fun projects.

Reply:
After a discussion with @jmaupetit, the current approach allows developers to overload this property and make their own choice if they want another set of roles to be able to edit content.

I believe it would be beneficial to introduce an opinionated method for defining who can edit content, ensuring a consistent choice across various fun projects.

Finally, we can refactor these default values when the feature is requested. I don't think it will break anything for library consumers if we move them to settings. The class API would remain unchanged.

Reply:
LGTM :+1: